### PR TITLE
Ems json menu decode

### PR DESCRIPTION
**Pull request by theus77**
New json menu decode (function and filter)

### CODE REVIEW

File: Entity/JsonMenu.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            $this->bySlugs[$slug] = $item;
+            if (isset($item['children'])) {
+                $this->recursiveWalk($item['children'], $slug . $this->glue);
+            }
+        }
+    }
+
+    public function getBySlug(string $slug): array
+    {
+        return $this->bySlugs[$slug] ?? null;
```
Comment (coppee) [inline Entity/JsonMenu.php]: Adding '?' (null) to the return type.

File: Entity/JsonMenu.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public function getSlug(string $id): string
+    {
+        return $this->slugs[$id] ?? null;
```
Comment (coppee) [inline Entity/JsonMenu.php]: Adding '?' (null) to the return type.

File: Entity/JsonMenu.php (hunk 1 above)
Comment (dameert) [reply Entity/JsonMenu.php]: would be better to return an empty array

File: Entity/JsonMenu.php (hunk 2 above)
Comment (dameert) [reply Entity/JsonMenu.php]: would be better to return an empty array

**Pull request merged**